controllers/apps/transformer_cluster_api_normalization.go

```diff
@@ -173,31 +169,6 @@ func (t *ClusterAPINormalizationTransformer) buildCompSpecs4Sharding(transCtx *c
 	return compSpecs, nil
 }
 
-func (t *ClusterAPINormalizationTransformer) buildCompLabelsInheritedFromCluster(transCtx *clusterTransformContext,
-	cluster *appsv1.Cluster) map[string]map[string]string {
-	clusterLabels := filterReservedLabels(cluster.Labels)
-	labels := make(map[string]map[string]string)
-	for _, compSpec := range transCtx.ComponentSpecs {
-		labels[compSpec.Name] = maps.Clone(clusterLabels)
-	}
-	for name, shardingCompSpecs := range transCtx.ShardingComponentSpecs {
-		for _, compSpec := range shardingCompSpecs {
-			labels[compSpec.Name] = controllerutil.MergeMetadataMaps(clusterLabels, constant.GetShardingNameLabel(name))
```

Comment:
The sharding component should include a label with the sharding name, as some scenarios (such as backup and recovery) will rely on this label for filtering and judgment.